lang/test/org/partiql/lang/eval/ExprValueFactoryTest.kt

```diff
@@ -365,4 +369,86 @@ class ExprValueFactoryTest {
+            add("timezone_hour", ion.newInt(totalSeconds?.div(3600)))
+            add("timezone_minute", ion.newInt(totalSeconds?.div((60))?.rem(60)))
+            addTypeAnnotation("\$partiql_time")
+        }
+
+    @Test
+    fun localTimeExprValueTest() {
+        val time = LocalTime.of(23, 2, 29, 23)
+        val timeExprValue = factory.newTime(time)
+        assertEquals(
+            expected = time,
+            actual = timeExprValue.scalar.timeValue(),
+            message = "Expected values to be equal."
+        )
+        val timeIonValue = timeExprValue.ionValue
+        val expectedIonValue = createIonTimeStruct(time.hour, time.minute, time.second, time.nano)
+        assertEquals(
+            expected = expectedIonValue,
+            actual = timeIonValue,
+            message = "Expected ionValues to be equal."
+        )
+    }
+
+    @Test
+    fun offsetTimeExprValueTest() {
+        val time = LocalTime.of(23, 2, 29, 23)
+        val zoneOffset = ZoneOffset.ofTotalSeconds(3600)
+        val offsetTime = OffsetTime.of(time, zoneOffset)
+        val timeExprValue = factory.newTime(time, zoneOffset)
+        assertEquals(
+            expected = offsetTime,
+            actual = timeExprValue.scalar.timeWithTimezoneValue(),
+            message = "Expected values to be equal."
+        )
+        val timeIonValue = timeExprValue.ionValue
+        val expectedIonValue = createIonTimeStruct(time.hour, time.minute, time.second, time.nano, zoneOffset.totalSeconds)
+        assertEquals(
+            expected = expectedIonValue,
+            actual = timeIonValue,
+            message = "Expected ionValues to be equal."
+        )
+    }
+
+    @Test
+    fun genericTimeExprValueTest() {
+        val timeExprValue = factory.newTime(23, 2, 29, 23, 2)
+        assertEquals(
+            expected = LocalTime.of(23, 2, 29),
+            actual = timeExprValue.scalar.timeValue(),
+            message = "Expected values to be equal."
+        )
+        val timeIonValue = timeExprValue.ionValue
+        val expectedIonValue = createIonTimeStruct(23, 2, 29)
+        assertEquals(
+            expected = expectedIonValue,
+            actual = timeIonValue,
+            message = "Expected ionValues to be equal."
+        )
+    }
+
+    @Test
+    fun genericTimeExprValueTest2() {
+        val timeExprValue = factory.newTime(23, 2, 29, 23, 2, -720)
+        assertEquals(
+            expected = OffsetTime.of(23, 2, 29, 0, ZoneOffset.ofTotalSeconds(-720*60)),
+            actual = timeExprValue.scalar.timeWithTimezoneValue(),
+            message = "Expected values to be equal."
+        )
+        val timeIonValue = timeExprValue.ionValue
+        val expectedIonValue = createIonTimeStruct(23, 2, 29, 0, -720*60)
+        assertEquals(
+            expected = expectedIonValue,
+            actual = timeIonValue,
+            message = "Expected ionValues to be equal."
+        )
+    }
```

Comment:
There is a lot of structurally similar code here--consider making this parameterized or at the least factor out the common code into a method.

Reply:
I have removed the `ionValue` assertion from here as those assertions are covered in `EvaluatingCompilerDateTimeTests`.